### ISSUE TITLE
'Error: ENOENT: no such file or directory' on starting sleek

### ISSUE THREAD

**Issue by mdbdb** [ransome1/sleek#687]
## 'Error: ENOENT: no such file or directory' on starting sleek

**App Version:** 2.0.11

**Platform:** Linux

**Installation Method:** App Store (Flatpak)

**Expected Behavior:**
Sleek opens the last opened file when it starts. All tasks are displayed.

**Actual Behavior:**
Sleek opens the last opened file when it starts. Sometimes I get this message: ENOENT: no such file or directory

When I start sleek via the console, I get these messages:

```
flatpak run com.github.ransome1.sleek
[2:0323/064608.180261:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: Datei oder Verzeichnis nicht gefunden`
Gtk-Message: 06:46:08.277: Failed to load module "canberra-gtk-module"
[2:0323/064608.298509:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: Datei oder Verzeichnis nicht gefunden
[2:0323/064608.298538:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: Datei oder Verzeichnis nicht gefunden
Styles injected found in CSS file: /home/rhs/.var/app/com.github.ransome1.sleek/config/sleek/userData/customStyles.css
[Error: ENOENT: no such file or directory, open '/run/user/1000/doc/8b60c039/todo.txt'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/run/user/1000/doc/8b60c039/todo.txt'
}
[Error: ENOENT: no such file or directory, open '/run/user/1000/doc/8b60c039/todo.txt'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/run/user/1000/doc/8b60c039/todo.txt'
}
[2:0323/064623.026280:ERROR:browser_main_loop.cc(276)] GLib-GObject: invalid cast from 'GtkFileChooserNative' to 'GtkWidget'
[2:0323/064626.041159:ERROR:browser_main_loop.cc(276)] GLib-GObject: ../gobject/gsignal.c:2778: instance '0x31ec019c02e0' has no handler with id '3194'
```

The tasks in my todo.txt are not displayed.

**Steps to Reproduce:**

The error occurs sporadically. I can't reproduce the error.

**Comment by ransome1** [comment]
@mdbdb thanks for reporting this. Can I ask you if you're using drag and drop in order to open the file in sleek? Or are you using the file chooser window?

**Comment by mdbdb** [comment]
> @mdbdb thanks for reporting this. Can I ask you if you're using drag and drop in order to open the file in sleek? Or are you using the file chooser window?

I've never used drag and drop to open my todo.txt file. I'm using the chooser window.

**Comment by ransome1** [comment]
It's really hard to say, it looks like something something sandbox. But I havn't able to reproduce this yet hence I cannot provide you solution at this point. 

In the meantime, did you stumble upon anything which can help me reproducing this?

**Comment by mdbdb** [comment]
No, I'm sorry. I don't know what causes the error and unfortunately I can't reproduce it.
But it still happens at irregular intervals.

**Comment by mdbdb** [comment]
About two weeks ago, I swapped the Flatpak version of Sleek for the AppImage version.
Since this change, the error has no longer occurred. It seems to be related to the Flatpak version.

**Comment by akademy** [comment]
This makes sense. I believe Flatpac does weird things with user access.

My experience is similar. Everytime I select my todo.txt file in my user directory e.g. `/home/user/todo.txt` it saves the file to the config file as `/run/user/1000/randomstring/todo.txt` - but the folder /run/user/1000 is temporary, and gets deleted on logout. So I get an error when it starts up, and have to reselect it.

I am not sure about the specifics though... 

I am on a Fedora linux and installed from the latest Flatpac version.

**Comment by akademy** [comment]
Can confirm that installing directly from a release version here on GitHub (i.e. https://github.com/ransome1/sleek/releases/download/v2.0.19/sleek-2.0.19-linux-x86_64.rpm) that the app then works as expected. The config file is storing the correct location for the todo.txt file.

**Comment by dubst3pp4** [comment]
I have the same problem with Sleek 2.0.16, installed via Flatpak on Fedora Silverblue 41. Is there a way to convince Sleek to use the `/home/user/...` paths instead the temporary ones? For example with some settings in Flatseal?

**Comment by Lezurex** [comment]
As @dubst3pp4 and @akademy mentioned, this is an issue related to the Flatpak sandbox. For files you want to access regularly, you need to give sleek the permission to do so on its own.

Use this command to give sleek access to any files you want to access regularly: `flatpak override --user com.github.ransome1.sleek --filesystem="/path/to/todo.txt"`

I added a wiki page explaining this for future reference: https://github.com/ransome1/sleek/wiki/Flatpak-Caveats
